asyncwrapper/src/main/kotlin/org/n/riesgos/asyncwrapper/process/wps/WPSProcess.kt

```diff
@@ -80,66 +84,104 @@ class WPSProcess(private val wpsClient : WPSClientSession, private val url: Stri
             val requestText = WPS20ExecuteEncoder.encode(executeRequest)
             LOGGER.info(requestText)
 
-            // @TODO: if falure due to network-problems, repeat n times before giving up.
-            val output = wpsClient.execute(url, executeRequest, wpsVersion)
+            var processOutput : ProcessOutput? = null;
+            //if failure due to network-problems, repeat n times before giving up.
 
-            var result: org.n52.geoprocessing.wps.client.model.Result =
-                if (output is org.n52.geoprocessing.wps.client.model.Result) {
-                    output
-                } else {
-                    (output as StatusInfo).result
-                }
+            if (this.retryConfig.attempts < 1 || this.retryConfig.backoff_millis < 1){
+                throw IllegalArgumentException("invalid retry configuration, values < 1 not allowed")
+            }
 
-            LOGGER.info("request result: ${result.toString()}")
-            val outputs: List<Data> = result.outputs
-            LOGGER.info("request outputs: ${outputs.toString()}")
-            var refOutputs = HashMap<String, MutableList<ReferenceParameter>>()
-
-            LOGGER.info("Start extracting results from the wps")
-            val setOutputIdentifiers = HashSet(expectedOutputs.map { it.wpsIdentifier})
-            for(expectedOutputIdentifier in setOutputIdentifiers){
-                val outputs = getOutputsById(expectedOutputIdentifier, outputs)
-                if(!outputs.isEmpty()){
-                    for (output in outputs) {
-                        val complexOutput = output.asComplexReferenceData()
-                        val format = complexOutput.format
-                        val schema = emptyStringIfNull(format.schema)
-                        val outputParam = ReferenceParameter(complexOutput.id, complexOutput.reference.href.toString(), format.mimeType, format.encoding, schema)
-                        if (!refOutputs.containsKey(complexOutput.id)) {
-                            refOutputs[complexOutput.id] = mutableListOf(outputParam)
-                        } else {
-                            refOutputs[complexOutput.id]!!.add(outputParam)
-                        }
-                        LOGGER.info("Stored result for " + complexOutput.id)
+            while(this.retryAttempts < this.retryConfig.attempts) {
+                try {
+                    this.retryAttempts++
```

Comment:
Honestly I would increase the retry attempts only in the code of the error handling. (Basically when I try it the first time, the retry counter should be 0. It should only increase when I got an error & decide that I want to retry).

I guess it is still working to have it here, but it feels a bit odd to me. (But I would also let myself be outvoted here)